Documentation/InjectionPatterns.md

```diff
@@ -40,22 +40,22 @@ class PetOwner: Person {
 }
 ```
 
-Note that the actual type of `Animal` passed to the initializer of `PetOwner` is automatically resolved by Swinject when the `PetOwner` instance is created:
+Note that the actual type of `Animal` passed to the initializer of `PetOwner` is automatically resolved by Swinject when the `PetOwner` instance is created.
 
 ## Property Injection
 
-_Property injection_ is a pattern to pass a dependency to a dependent instance via a setter property. Property injection is appropriate if the dependency is optional to the dependent instance.
+_Property injection_ is a pattern to pass a dependency to a dependent instance via a setter property. Property injection is appropriate if the dependency is optional to the dependent instance, and is required in the case of _[circular dependencies](CircularDependencies.md)_.
 
 The following code defines property injection to `PetOwner2`:
 
 ```swift
 let container = Container()
 container.register(Animal.self) { _ in Cat() }
-container.register(Person.self) { r in
-    let owner = PetOwner2()
-    owner.pet = r.resolve(Animal.self)
-    return owner
-}
+container.register(Person.self) { _ in PetOwner2() }
+    .initCompleted { r, p in
+        let owner = p as! PetOwner2
+        owner.pet = r.resolve(Animal.self)
+    }
```

Comment:
Original registration example should be prefered, as it is simple and does not require casting.

Reply:
however once you have circular dependencies, you'll have to refactor it as mentioned in the circular dependencies doc

Reply:
Not necessarily - you would need to refactor only one definition (in the case of 2 related objects).

Reply:
right, the docs mention you should change to `initCompleted` "in either or both". That was the keypoint that made me think I should open a PR (if always using it has no problems, and never using it may run into problems, then I would choose to always use it).

But if it's not the preferred way, and users should remember to switch to `initCompleted` only once they have problems, then I think this PR doesn't make much sense. Sorry about that!